### ISSUE TITLE
Starting translating then Failed iniOS

### ISSUE THREAD

**Issue by Frenzyfeng** [facebook/infer#824]
**First i do the clean and it success:**
xcodebuild -workspace XiaoYing.xcworkspace -configuration Debug -scheme XiaoYing clean

**Then i do this**:
 infer --keep-going -- xcodebuild -workspace XiaoYing.xcworkspace -configuration Debug -scheme XiaoYing

The log tell me '▸ Build Succeeded'

**But then i got all error like thie**: 

Starting translating 2066 files
error: error reading '/Users/ios/Library/Developer/Xcode/DerivedData/XiaoYing-bmlycmlzyvjnuwfsrfpsfxbgcgvg/Build/Index/DataStore'
error: error reading '/Users/ios/Library/Developer/Xcode/DerivedData/XiaoYing-bmlycmlzyvjnuwfsrfpsfxbgcgvg/Build/Index/DataStore'
1 error generated.
1 error generated.
ERROR RUNNING CAPTURE: Bi_inbuf.End_of_input
Raised at file "src/bi_inbuf.ml", line 49, characters 6-24
Called from file "src/bi_io.ml" (inlined), line 508, characters 12-35
Called from file "atd/clang_ast_b.ml", line 27505, characters 7-24
Called from file "clang/Capture.ml", line 48, characters 8-39
Called from file "clang/Capture.ml" (inlined), line 93, characters 6-35
Called from file "clang/Capture.ml", line 157, characters 26-73

Error: the following clang command did not run successfully:
  '/usr/local/Cellar/infer/0.13.0/lib/infer/facebook-clang-plugins/clang/install/bin/clang' "@/var/folders/_0/fnvrj35x6_qfffps9kvf2v140000gn/T/clang_command_acbd76.txt"
error: error reading '/Users/ios/Library/Developer/Xcode/DerivedData/XiaoYing-bmlycmlzyvjnuwfsrfpsfxbgcgvg/Build/Index/DataStore'

**And finally i got nothing**:

Nothing to compile. Try running `xcodebuild -workspace XiaoYing.xcworkspace -configuration Debug -scheme XiaoYing clean` first.
There was nothing to analyze.

How to solve this problem?

**Comment by dulmarod** [comment]
The main problem seems to be:

error: error reading '/Users/ios/Library/Developer/Xcode/DerivedData/XiaoYing-bmlycmlzyvjnuwfsrfpsfxbgcgvg/Build/Index/DataStore'

Do you know what that could be? Does the compilation command succeed on its own (without Infer)?

**Comment by Frenzyfeng** [comment]
@dulmarod Yes I tried the compilation command without infer and it exactly succeed.

I have no idea with this : error: error reading '/Users/ios/Library/Developer/Xcode/DerivedData/XiaoYing-bmlycmlzyvjnuwfsrfpsfxbgcgvg/Build/Index/DataStore'

**Comment by Mr-Crh** [comment]
you can try to look up the issue #759 and using command:
infer --keep-going --no-xcpretty -- xcodebuild -workspace XiaoYing.xcworkspace -configuration Debug -scheme XiaoYing

**Comment by Mr-Crh** [comment]
but i guess you might have problems later about build failed:
The following build commands failed:
Ld /Users/***/Library/Developer/Xcode/DerivedData...........
and then you can look up the issue #810 and using command:
LD=/usr/bin/true infer --keep-going --no-xcpretty -- xcodebuild -workspace XiaoYing.xcworkspace -configuration Debug -scheme XiaoYing

best wish~

**Comment by Frenzyfeng** [comment]
@Mr-Crh Thanks for your answer, But after i tried #759 &  #810 command, i still have problem.

**As #810 mentioned, i used :**

export LD=/usr/bin/true infer
export CreateUniversalBinary=/usr/bin/true

**Then i got clean success**: 

xcodebuild clean -scheme appName -workspace appName.xcworkspace -configuration Debug

**And then** :

LD=/usr/bin/true infer --keep-going --no-xcpretty -- xcodebuild -workspace XiaoYing.xcworkspace -configuration Debug -scheme XiaoYing

**Finally i got error:** 


CreateUniversalBinary Build/Products/Debug-iphoneos/XiaoYing.app/XiaoYing normal armv7\ arm64
    cd /Users/ios/Documents/Scan/iOSVivaVideo/XiaoYingApp
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/bin:/bin:/bin:::::/usr/local/Cellar/jekyll:/Users/ios/.rvm/gems/ruby-2.4.0/bin:/Users/ios/.rvm/gems/ruby-2.4.0@global/bin:/Users/ios/.rvm/rubies/ruby-2.4.0/bin:/bin:/bin:/bin:::::/usr/local/Cellar/jekyll:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ios/.rvm/bin:pwd/infer/infer/bin:pwd/infer/infer/bin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo -create /Users/ios/Documents/Scan/iOSVivaVideo/XiaoYingApp/Build/Intermediates/XiaoYing.build/Debug-iphoneos/XiaoYing.build/Objects-normal/armv7/XiaoYing /Users/ios/Documents/Scan/iOSVivaVideo/XiaoYingApp/Build/Intermediates/XiaoYing.build/Debug-iphoneos/XiaoYing.build/Objects-normal/arm64/XiaoYing -output /Users/ios/Documents/Scan/iOSVivaVideo/XiaoYingApp/Build/Products/Debug-iphoneos/XiaoYing.app/XiaoYing
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't open input file: /Users/ios/Documents/Scan/iOSVivaVideo/XiaoYingApp/Build/Intermediates/XiaoYing.build/Debug-iphoneos/XiaoYing.build/Objects-normal/armv7/XiaoYing (No such file or directory)

** BUILD FAILED **


The following build commands failed:
	CreateUniversalBinary Build/Products/Debug-iphoneos/XiaoYing.app/XiaoYing normal armv7\ arm64
(1 failure)

Error backtrace:
Raised at file "integration/Driver.ml", line 297, characters 16-78
Called from file "integration/Driver.ml", line 176, characters 2-16
Called from file "backend/infer.ml", line 23, characters 2-36
Called from file "backend/infer.ml", line 106, characters 6-52
Uncaught error: ("InferModules__Driver.Infer_error(\"exited with code 65\")")

**Comment by Mr-Crh** [comment]
@Frenzyfeng  you can try to use the following command after you clean the workspace:
infer run -- xcodebuild -workspace  XiaoYing.xcworkspace -scheme XiaoYing
and then, it may be prompted "Build Succeeded" and "Starting translating *** files" and a lot of error such as:
error: error reading '/Users/****/Library/Developer/Xcode/DerivedData/*********/Index/DataStore'
1 error generated.
1 error generated.
1 error generated.
1 error generated.
ERROR RUNNING CAPTURE: Bi_inbuf.End_of_input
Raised at file "src/bi_inbuf.ml", line 49, characters 6-24
Called from file "src/bi_io.ml" (inlined), line 508, characters 12-35
Called from file "atd/clang_ast_b.ml", line 27505, characters 7-24
Called from file "clang/Capture.ml", line 48, characters 8-39
Called from file "clang/Capture.ml" (inlined), line 93, characters 6-35
Called from file "clang/Capture.ml", line 157, characters 26-73

now use command"LD=/usr/bin/true infer --keep-going --no-xcpretty -- xcodebuild -workspace XiaoYing.xcworkspace -configuration Debug -scheme XiaoYing".

that's all, you can try it~

**Comment by Frenzyfeng** [comment]
@Mr-Crh That really work for me.
Thank you so~~~o mush and be~~~st wishes to you.

**Comment by Mr-Crh** [comment]
@Frenzyfeng You are welcome :)

**Comment by cyw3** [comment]
I have the same problem but But after i tried #759 & #810 and this issue's command, i still have problem.

LD=/usr/bin/true infer run --keep-going --no-xcpretty -- xcodebuild build -scheme appName -configuration Debug -sdk iphonesimulator

![image](https://user-images.githubusercontent.com/11549103/37906434-ac8eed0a-3134-11e8-9657-d8bdeaf69141.png)
